File: fix_unused_imports.py
#!/usr/bin/env python3
import os
import glob
import re

def fix_unused_imports(file_path):
    """Remove unused imports from page files"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has unused ArrowRight or Helmet imports
        if 'ArrowRight' in content and 'ArrowRight' not in content[content.find('return'):]:
            # Remove ArrowRight from imports
            content = re.sub(r',\s*ArrowRight', '', content)
            content = re.sub(r'ArrowRight\s*,?\s*', '', content)
        
        if 'Helmet' in content and 'Helmet' not in content[content.find('return'):]:
            # Remove Helmet from imports
            content = re.sub(r',\s*Helmet', '', content)
            content = re.sub(r'Helmet\s*,?\s*', '', content)
        
        # Clean up any double commas or trailing commas
        content = re.sub(r',\s*,', ',', content)
        content = re.sub(r',\s*}', '}', content)
        content = re.sub(r'{\s*,', '{', content)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        return True
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    # Find all page.tsx files
    page_files = glob.glob('app/**/page.tsx', recursive=True)
    
    fixed_count = 0
    for file_path in page_files:
        if fix_unused_imports(file_path):
            fixed_count += 1
    
    print(f"Fixed unused imports in {fixed_count} page files")

if __name__ == "__main__":
    main()

File: fix-unused-imports.js
import fs from 'fs';
import { glob } from 'glob';

// Function to remove unused imports from a file
function fixFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Split content into lines
    const lines = content.split('\n');
    const newLines = [];
    const usedImports = new Set();
    
    // First pass: find all used imports by scanning the content
    const contentText = content.toLowerCase();
    
    // Check for common component usage patterns
    const componentPatterns = [
      'navigation',
      'footer',
      'errorboundary',
      'error boundary'
    ];
    
    for (const pattern of componentPatterns) {
      if (contentText.includes(pattern)) {
        usedImports.add(pattern);
      }
    }
    
    // Second pass: filter imports
    for (let i = 0; i < lines.length; i++) {
      const line = lines[i];
      const trimmedLine = line.trim();
      
      // Check if this is an import line
      if (trimmedLine.startsWith('import ')) {
        // Extract the import name
        const importMatch = trimmedLine.match(/import\s+(\w+)\s+from/);
        if (importMatch) {
          const importName = importMatch[1].toLowerCase();
          
          // Check if this import is actually used
          if (usedImports.has(importName) || 
              contentText.includes(`<${importName}`) || 
              contentText.includes(`<${importName} `) ||
              contentText.includes(`<${importName}>`) ||
              contentText.includes(`</${importName}>`)) {
            newLines.push(line);
          } else {
            // Remove unused import
            modified = true;
            console.log(`Removing unused import: ${importName} from ${filePath}`);
          }
        } else {
          newLines.push(line);
        }
      } else {
        newLines.push(line);
      }
    }
    
    if (modified) {
      fs.writeFileSync(filePath, newLines.join('\n'));
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Main function to process files
async function main() {
  // Find all page files
  const files = await glob('app/**/page.tsx');

  console.log(`Found ${files.length} page files to process...`);

  let fixedCount = 0;
  files.forEach(file => {
    if (fixFile(file)) {
      fixedCount++;
    }
  });

  console.log(`Fixed ${fixedCount} files`);
}

main().catch(console.error);